esphome/components/esp32/post_build.py.script

```
@@ -121,7 +124,32 @@ def esp32_copy_ota_bin(source, target, env):
     new_file_name = env.subst("$BUILD_DIR/${PROGNAME}.ota.bin")
     shutil.copyfile(firmware_name, new_file_name)
     print(f"Copied firmware to {new_file_name}")
+    # Create MD5 checksum for OTA binary
+    create_ota_bin_md5(new_file_name)
 
+def create_ota_bin_md5(original_file_name):
+    """
+    Calculate an md5 hash for the specified file, storing in {original_file_name}.md5 .
+    """
+    file_path = pathlib.Path(original_file_name)
+    if not file_path.exists():
+        print(f"Warning: File {original_file_name} not found - skipping MD5 calculation")
+        return
```

not sure what catastrophic failure would do to the build but we could throw an error here if that's preferred.